module/x/gravity/keeper/msg_server.go

```diff
@@ -254,10 +254,33 @@ func (k msgServer) SendToEthereum(c context.Context, msg *types.MsgSendToEthereu
 
 // RequestBatchTx handles MsgRequestBatchTx
 func (k msgServer) RequestBatchTx(c context.Context, msg *types.MsgRequestBatchTx) (*types.MsgRequestBatchTxResponse, error) {
-	// TODO: limit this to only orchestrators and validators?
 	ctx := sdk.UnwrapSDKContext(c)
 	params := k.GetParams(ctx)
 
+	// Check sender is either from a validator or orchestrator
+	isAuthorized := false
+	delegateKeys := k.getDelegateKeys(ctx)
+	for _, delegateKey := range delegateKeys {
```

Comment:
will there be many delegate keys / keep increasing?

Reply:
I'm wondering if we should clean up the `delegatekeys` for every block at `BeginBlocker` so that we could correctly update the valset in gravity module. I didn't see any method for gravity module to delete `delegatekeys`.

https://github.com/crypto-org-chain/gravity-bridge/blob/f1b1ef0a7593695f4d0532cdb2da7f66970200a0/module/x/gravity/abci.go#L19-L25

Reply:
yeah but it should be fine in our case, for the time being, as we are not planning to have many update in our validator set. 
It goes along with the ability to update the delegateKeys once set which is a pending issue 
https://github.com/crypto-org-chain/gravity-bridge/issues/111